packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -465,12 +465,26 @@ export default declare((api, opts) => {
         transpileEnum(path, t);
       },
 
-      TSImportEqualsDeclaration(path) {
-        throw path.buildCodeFrameError(
-          "`import =` is not supported by @babel/plugin-transform-typescript\n" +
-            "Please consider using " +
-            "`import <moduleName> from '<moduleName>';` alongside " +
-            "Typescript's --allowSyntheticDefaultImports option.",
+      TSImportEqualsDeclaration(path: NodePath<t.TSImportEqualsDeclaration>) {
+        if (t.isTSExternalModuleReference(path.node.moduleReference)) {
+          // import alias = require('foo');
+          throw path.buildCodeFrameError(
+            `\`import ${path.node.id.name} = require('${path.node.moduleReference.expression.value}')\` ` +
+              "is not supported by @babel/plugin-transform-typescript\n" +
+              "Please consider using " +
+              `\`import ${path.node.id.name} from '${path.node.moduleReference.expression.value}';\` alongside ` +
+              "Typescript's --allowSyntheticDefaultImports option.",
+          );
+        }
+
+        // import alias = Namespace;
+        path.replaceWith(
+          t.variableDeclaration("var", [
+            t.variableDeclarator(
+              path.node.id,
+              entityNameToExpr(path.node.moduleReference),
+            ),
+          ]),
```

Comment:
Could you add a test case (that should fail) for:
```ts
var foo = 10
import Bar = foo
```

Reply:
Since `tsc` reports a type error here, if throwing from our side is too hard (we are not doing a good job when it comes to scope-tracking TS stuff) we can just always compile `import Bar = foo` to `var Bar = foo` regardless of what `foo` is.

Reply:
Actually, we probably cannot throw because in this case we don't know if `X` is a namespace:
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgDTgXzgMyhEcBEAnvgNwCwAUKJLHAEICGUcAvCuRY1AHQBiEEEkA

Reply:
May an assumption be helpful?

Reply:
An assumption helps making the code smaller or faster, which isn't the case here.

Reply:
Isn't this partially non-spec-compliant however?

Reply:
Actually, transforming `import Alias = foo` to `var Alias = foo` is exactly what `tsc` is doing. Try this out:

```ts
// index.ts
// @ts-ignore
import Foo from 'foo';
import Alias = Foo.A;
const a = Alias.a;
```

with:
```bash
tsc index.ts --module esnext
```

we got:

```js
// @ts-ignore
import Foo from 'foo';
var Alias = Foo.A;
var a = Alias.a;
```

`tsc` doesn't know if `Foo.A` is a namespace or not either.

Reply:
Probably it knows that it is not a namespace but it doesn't complain😄